src/ledger-iframe-bridge.test.ts

```diff
@@ -526,14 +544,19 @@ describe('LedgerIframeBridge', function () {
       removeEventListenerSpy = jest.spyOn(global.window, 'removeEventListener');
       addEventListenerSpy = jest.spyOn(global.window, 'addEventListener');
       bridge = new LedgerIframeBridge();
-      await bridge.init();
-      await simulateIFrameLoad(bridge.iframe);
+      await initializeBridge(bridge);
     });
 
     describe('when configurate bridge url', function () {
       describe('when given bridge url is different with current', function () {
         beforeEach(async () => {
-          await bridge.setOptions({ bridgeUrl: 'https://metamask.io' });
+          // since the bridge will be re-initialized, we need to
+          // simulate the iframe loading again
+          const setOptionHandle = bridge.setOptions({
+            bridgeUrl: 'https://metamask.io',
+          });
+          await simulateIFrameLoad(bridge.iframe);
+          await setOptionHandle;
```

Comment:
This does not work yet, but I noticed that `destroy` (which is called by `setOptions`) does not remove the iframe from the document - this means that if we call `destroy` and `init` again (like in `setOptions` implementation) then we'll end up with two iframes in the document. I'm not sure this is an issue that could cause some of the open issues in this repo - besides this it should be the reason why it is troublesome to simulate the iframe load in this case